microceph/ceph/config.go

```diff
@@ -3,15 +3,140 @@ package ceph
 import (
 	"context"
 	"database/sql"
+	"encoding/json"
 	"fmt"
 	"os"
 	"path/filepath"
 	"strings"
 
+	"github.com/canonical/microceph/microceph/api/types"
 	"github.com/canonical/microceph/microceph/common"
 	"github.com/canonical/microceph/microceph/database"
 )
 
+type ConfigExtras struct{
```

Comment:
Some docs on this type would be good, same for ConfigTable below.

I feel like the name of these two type could be clarified as well. Afaict the purpose of this type is to find services to restart when configs are set, so maybe `ServiceConfigs` and  `ServiceRestartMap`  or such? Possibly the types could be unified too

Reply:
Hmm, My inclination towards using "ConfigTable" was that not only is this particular Map the table for finding services to restart, this is also THE table which indexes all the configs we support being configured. It is then used in various ways (the slice of keys is used to filter configs..) In that essence, using ConfigTable (or ConfigMap) tells that this is the sole Map one needs to be concerned about when looking for configs.

Reply:
I agree with type unification, I have convoluted the ConfigExtras type into the Table itself.

Reply:
@sabaini thoughts about the naming dilemma ?

Reply:
Thanks for the clarification -- agree if it's slated to be used more generically the generic `ConfigTable` works better.

Maybe could add your thoughts about this type as a doc comment ;-)